test-vectors/credentials/create.json

```diff
@@ -552,16 +552,38 @@
     {
       "description": "creates a verifiable credential as a jwt with a did:key",
       "input": {
-        "signerDidUri": "did:key:zQ3shNLt1aMWPbWRGa8VoeEbJofJ7xJe4FCPpDKxq1NZygpiy",
-        "signerPrivateJwk": {
-          "kty": "EC",
-          "d": "OeBCt3M8roz9F9Ny192TtZXf_-qPzAkwSM6ep7QdSO4",
-          "use": "sig",
-          "crv": "secp256k1",
-          "kid": "NB_qlUIr06-AikVVMFreq0lc-omQtzc6lwhhcvgO6r4",
-          "x": "DdtN8W6x_34pB_nkxR0e1tmDkNnsJeusBAEPzKWgf_Y",
-          "y": "u3W135inodLqtcEb9jNGS3JsM_uFKmkJSb8Trc9luWI",
-          "alg": "ES256K"
+        "signerPortableDid": {
```

Comment:
I am hesitant to make this change because I am not sure portable did is a concept that should be exposed in the spec.

the spec serves mainly to facilitate interop, and the serialization format of DIDs seems to be an internal feature to the SDKs. if we have a use case to enable interoperability via portable DID that would help change my mind here.

Reply:
That's fair... I made this change because this test is about importing an existing did, signing a VC with the equivalent bearer did, and checking that the vcjwt created is what we expect. 

In your view, how could we test this aspect of credential creation but still keep interop top of mind?

Reply:
We could still define input keys (or key seeds) and expected outputs without having a common bearer did structure.

The DID working group argued about a similar concept for a while and agreed to not define a serializable representation of private keys as to avoid recommending doing it, since it's a security risk to transmit private keys.

Some DID methods do provide private keys. Others provide public keys and test for common identifier generation and document expansion, like did key: https://w3c-ccg.github.io/did-method-key/#ed25519-x25519

Reply:
hmm... it seems like you may not be onboard with the idea of `PortableDid` existing in general, because its purpose is basically to transmit private key (imo)?

if that's the case, then we need to have a bigger discussion around why portabledid should or should not exist. i'd appreciate it if you could raise it as a ticket in this repo and document your concern around it.

as far as this test vector is concerned, would you feel more comfortable if i got rid of the `signerPortableDid` key, and have `uri`, `privateKeys`, `document` directly under `input` key? then it's not really introducing the concept of portabledid to the test vectors, but allows the test vector consumer to still have the necessary info to create a did they can then sign the VC with to produce the same expected `vector.output` vcJwt?

Reply:
fixed in https://github.com/TBD54566975/web5-spec/pull/130/commits/78663ce3b912e600fff4099543da55c3ffc477fb

Reply:
I think we probably have a number of existing test vectors which have implementation-specific assumptions, such as the concept of a "portable DID" so even under the working assumption "portable DID" is not something to be within the web5 spec (that's my working stance) I think it pragmatic to expect to commit to a work item at some point to twease apart our test vectors for vectors which are spec-specific vs our-implementations-specific

TL;DR I agree `test-vectors/portable_did/*` shouldn't be where it is but we can come back later b/c we need to review all test vectors

Reply:
hmm, so there seems to be a couple opinions emerging on what test vectors should do -

1. allow for interop with external repos that want to make sure their way of building and using DIDs and VCs is consistent with ours (this comment)

2. allow for interop between _our per-language sdk implementations_ to make sure we are all building and using DIDs and VCs consistent with (each other) [this comment](https://github.com/TBD54566975/web5-kt/pull/262#discussion_r1537824348)

i think it should do both, which is why i pulled out the `portableDid` concept out of this test vector for VC creation, and created a new `test-vectors/portable_did/parse.json` test vector for parsing in a portable did and importing into a bearerdid.

Reply:
@decentralgabe 

>  if we have a use case to enable interoperability via portable DID that would help change my mind here.

we do. right now across our sdks, you can create port a did around using `PortableDid`. trying to better understand how inclusion of `portableDid` prevents interop. couldn't third parties just ignore the "portableDid" property and go straight to private keys if desired? asking because leaving it out makes it more confusing / a bit awkward when consuming the vectors in our own sdks

Reply:
i think this needs a bit more discussion - reverted back the changes to make it easier for other sdks to consume the portabledid when writing tests against this vector - https://github.com/TBD54566975/web5-spec/issues/142

Reply:
> you can port a did around using PortableDid

I believe this has security risks and we should not support it, though I acknowledge it's useful for test vectors. Happy to continue the discussion in #142.